lib/paginator/config.ex

```diff
@@ -31,17 +31,40 @@ defmodule Paginator.Config do
       after_values: Cursor.decode(opts[:after]),
       before: opts[:before],
       before_values: Cursor.decode(opts[:before]),
-      cursor_fields: opts[:cursor_fields],
+      cursor_fields: opts[:cursor_fields] || [],
       include_total_count: opts[:include_total_count] || false,
-      total_count_primary_key_field: opts[:total_count_primary_key_field] || @default_total_count_primary_key_field,
+      total_count_primary_key_field:
+        opts[:total_count_primary_key_field] || @default_total_count_primary_key_field,
       limit: limit(opts),
-      sort_direction: opts[:sort_direction] || :asc,
+      sort_direction: opts[:sort_direction],
       total_count_limit: opts[:total_count_limit] || @default_total_count_limit
     }
+    |> convert_deprecated_config()
   end
 
   defp limit(opts) do
     max(opts[:limit] || @default_limit, @minimum_limit)
     |> min(opts[:maximum_limit] || @maximum_limit)
   end
+
+  defp convert_deprecated_config(config) do
+    case {config, Keyword.keyword?(config.cursor_fields)} do
+      {%__MODULE__{sort_direction: _}, true} ->
+        config
+
+      {%__MODULE__{sort_direction: nil}, false} ->
+        %{config | cursor_fields: make_keyword_list(config.cursor_fields, :asc)}
+
+      {%__MODULE__{sort_direction: direction}, false} ->
+        %{
+          config
+          | cursor_fields: make_keyword_list(config.cursor_fields, direction),
+            sort_direction: nil
+        }
+    end
+  end
+
+  defp make_keyword_list(fields, sorting_direction) do
```

Comment:
Nitpicking here but what about renaming this to: `build_cursor_fields_from_sort_direction`? I feel like it would be a bit more descriptive.